kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/DiscoveryClientProvider.scala

```diff
@@ -42,4 +41,15 @@ object DiscoveryClientProvider extends Logging {
     }
   }
 
+  def createDiscoveryClient(conf: KyuubiConf): DiscoveryClient = {
+    val className = conf.get(KyuubiConf.DISCOVERY_CLIENT_CLASS)
+    val clazz =
+      try {
+        Class.forName(className)
+      } catch {
+        case ex: Throwable => throw new KyuubiException(s"Class not found $className", ex)
+      }
+    val constructor = clazz.getConstructor(conf.getClass)
+    constructor.newInstance(conf).asInstanceOf[DiscoveryClient]
```

Comment:
Another check is required here, please refer `AuthenticationProviderFactory.getAuthenticationProviderForServer`

Reply:
Thanks for the suggestion! Updated the PR.